apps/prairielearn/assets/scripts/instructorFileEditorClient.ts

```diff
@@ -117,6 +155,36 @@ class InstructorFileEditor {
     this.checkDiff();
     this.editor.resize();
   }
+
+  reformatJSONFile() {
+    const contents = this.editor.getValue();
+    const cursorOffset = getCursorOffsetFromCursorPosition(
+      this.editor.getCursorPosition(),
+      this.editor.getSession().getDocument().getAllLines(),
+    );
+    prettier
+      .formatWithCursor(contents, {
+        cursorOffset,
+        parser: 'json',
+        plugins: [prettierBabelPlugin, prettierEstreePlugin],
+      })
+      .then(({ formatted, cursorOffset }) => {
```

Comment:
Any reason why this function can't be made async? It is completing its tasks asynchronously anyway...

Reply:
We have to use `.then()`/`.catch()` *somewhere* since this is being invoked from an even handler. If you'd prefer me to lift that up to the event handler, I can make that change.

Reply:
Done in 82b7e00.

Reply:
> We have to use `.then()`/`.catch()` _somewhere_ since this is being invoked from an even handler.

Technically you don't, you can have an async event handler, the only thing is that the event handler will return (the promise) before it completes its action, which technically already happens anyway with a then/catch. See:

https://github.com/PrairieLearn/PrairieLearn/blob/7457bcefff2e79c5e1d9a797721b7e0b8e878a4a/apps/prairielearn/assets/scripts/instructorAssessmentManualGradingAssessmentQuestionClient.ts#L28
https://github.com/PrairieLearn/PrairieLearn/blob/7457bcefff2e79c5e1d9a797721b7e0b8e878a4a/apps/prairielearn/assets/scripts/instructorAssessmentManualGradingAssessmentQuestionClient.ts#L240

Reply:
I know you can *technically* pass an async function there, but `addEventListener` and basically all similar APIs won't handle errors (Promise rejections). I consider it a best practice to always explicitly handler errors in this case, exactly the same as we would on the server in similar situations.